mage_integrations/mage_integrations/sources/sql/base.py

```diff
@@ -232,11 +232,14 @@ def __fetch_rows(
         order_by_columns = []
 
         # This order is very important, don’t change
-        for arr in [
-            bookmark_properties,
+        order_by_properties = [
```

Comment:
There're some issues here.
1. When we order the table by key_properties, we need to use that key as bookmark properties as well because the original bookmark properties will not be logged in order.
2. After the first sync, we need to fetch the max values of the original bookmark properties and then update the state file for future incremental sync.
3. If there're no key properties, we still need to order the rows by bookmark properties.
4. Regarding performance differences, we tested on a dataset with 16 million rows. Ordering by key columns and fetching sub batch took 30 seconds while ordering by a non-key columns took 3~4 minutes. So i think this strategy will help speed up syncing large datasets.

There're some non-trivial corner cases. Let me know if anything is unclear.